### DISCUSSION TITLE
IgnoreDifferences on partial image string value

### DISCUSSION THREAD

**Discussion by JonathanSerafini** [argoproj/argo-cd#16296] [Q&A]
I'm curious to see if it would be possible to IgnoreDifferences on our applications, in such a away where
we only ignore changes to a part of the image string, specifically the `@sha256` portion of the image.

Specifically, I'd like to consider the entire image before an _optional_ `@sha256:xxx` as valid for diff
detection, and ignore everything after.

Currently, our CI pipelines are updating the application manifests to push out image changes
in a `registry/image:tag` format.

We are also looking at deploying the Sigstore policy controller which contains a validating/mutating webhook
that appends the shasum to the images so that the end result is an image in `registry/image:tag@sha256:xxx`
format.

With the apps set to automated syncPolicy, we'd end up with constant diffs and looping updates.

**Comment by rsmartins78** [comment]
I got the same issue here with Binary Authorization on GKE. The sha256 is appended to the `image` and ArgoCD then marks the app as out of sync.
![Screenshot 2025-06-03 at 14 21 53](https://github.com/user-attachments/assets/7f8f485a-522b-4409-a5b2-36f4b50c4382)
Did anyone got a solution for that?
`ArgoCD Version: v3.0.1+2bcef48`

**Reply by crenshaw-dev** [reply]
Is it added by a mutating webhook? If so, Server-side-diff might handle this gracefully.

**Reply by rsmartins78** [reply]
Server side apply caused a different issue with cloudsql auth proxy operator, so I just stopped using the binary authorization mutation hook as it wasn’t in prod anyway

**Comment by cme-incom** [comment]
I’m experiencing the same issue.
Has anyone found a workaround or fix yet?
